WordPress/src/main/java/org/wordpress/android/datasets/ReaderBlogTable.java

```diff
@@ -90,10 +92,36 @@ public static ReaderBlog getBlogInfo(long blogId, String blogUrl) {
         }
     }
 
-    public static String getBlogUrl(long blogId) {
-        String[] args = {Long.toString(blogId)};
-        return SqlUtils.stringForQuery(
-                ReaderDatabase.getReadableDb(), "SELECT blog_url FROM tbl_blog_info WHERE blog_id=?", args);
+    public static long getBlogIdFromUrl(String url) {
+        if (TextUtils.isEmpty(url)) {
+            return 0;
+        }
+        String normUrl = UrlUtils.normalizeUrl(url);
+        String[] args = {normUrl, normUrl};
+        return SqlUtils.longForQuery(ReaderDatabase.getReadableDb(),
+                "SELECT blog_id FROM tbl_blog_info WHERE feed_url=? OR blog_url=?",
```

Comment:
Wonder why we need to check the `feed_url` URL here.  Same consideration on line 113, where there is a check for `blog_url` instead.